lang/llvm-12/Portfile

```diff
@@ -0,0 +1,306 @@
+                        target-independent optimizer, along with code \
+                        generation support for many popular CPUs (as well as \
+                        some less common ones!) These libraries are built \
+                        around a well specified code representation known as \
+                        the LLVM intermediate representation ("LLVM IR").
+
+    depends_lib-append  port:libedit port:libffi port:ncurses path:lib/libxar.dylib:xar port:zlib
+    depends_run-append  bin:perl:perl5 port:llvm_select
+
+    configure.args-append \
+        -DLLVM_ENABLE_PROJECTS="polly"
+}
+
+if {${subport} eq "clang-${llvm_version}"} {
+
+    homepage            https://clang.llvm.org/
+    description         C, C++, Objective C and Objective C++ compiler
+    long_description    Clang is an "LLVM native" C/C++/Objective-C compiler, \
+                        which aims to deliver amazingly fast compiles (e.g. \
+                        about 3x faster than GCC when compiling Objective-C \
+                        code in a debug configuration), extremely useful error \
+                        and warning messages and to provide a platform for \
+                        building great source level tools. The included Clang \
+                        Static Analyzer is a tool that automatically finds bugs in \
+                        your code, and is a great example of the sort of tool \
+                        that can be built using the Clang frontend as a \
+                        library to parse C/C++ code.
+
+    depends_lib-append  port:libedit port:libffi port:ncurses path:lib/libxar.dylib:xar port:zlib port:llvm-${llvm_version}
+    depends_run-append  bin:perl:perl5 port:clang_select
+
+    # not yet tested and patches need to be moved
+    if {${os.platform} eq "darwin" && ${os.major} < 17} {
+        depends_build
+        depends_lib
+        depends_run
+        archive_sites
+        known_fail yes
+        pre-fetch {
+            ui_error "${subport} is not supported on this os version at present."
+            return -code error {unsupported platform}
+        }
+    }
+
+    # CMAKE_LINKER is used to determine the value for HOST_LINK_VERSION
+    configure.args-append \
+        -DCMAKE_LINKER=${prefix}/bin/ld    \
+        -DCLANG_INCLUDE_TESTS=OFF          \
+        -DCLANG_ENABLE_STATIC_ANALYZER=OFF \
+        -DCLANG_ENABLE_ARCMT=OFF           \
+        -DDARWIN_PREFER_PUBLIC_SDK=ON      \
+        -DLLVM_BUILD_RUNTIME=ON            \
+        -DLIBCXX_ENABLE_SHARED=ON          \
+        -DLIBCXX_INSTALL_LIBRARY=ON
+
+    # clang project requires libcxx project
+    # else the built 1st stage compiler tries to link against libstdc++ and finds an old version
+    # flang project requires mlir project
+    configure.args-append   \
+        -DLLVM_ENABLE_PROJECTS="clang\;clang-tools-extra\;compiler-rt\;libcxx\;libcxxabi\;flang\;mlir\;lld\;openmp\;libunwind"
```

Comment:
I'll take openmp out of here, as we use an external build of openmp and hack the clang code to find it instead of this one anyway